src/perform_step/SROCK_perform_step.jl

```diff
@@ -425,7 +425,8 @@ end
 
       for i in 1:length(W.dW)
         WikJ = W.dW[i]; WikJ2 = vec_χ[i]
-        WikRange .= 1//2 .* (W.dW .* WikJ .- (1:length(W.dW) .== i) .* abs(dt) )#.+ (1:length(W.dW) .< i) .* dt .* WikJ2 .- (1:length(W.dW) .> i) .* dt .* vec_χ)
+        dwrange = 1:length(W.dW)
+        @.. WikRange = 1//2 * (W.dW * WikJ - (dwrange == i) * abs(dt))#.+ (1:length(W.dW) .< i) .* dt .* WikJ2 .- (1:length(W.dW) .> i) .* dt .* vec_χ)
```

Comment:
Maybe that's not necessary but you could avoid recomputing `abs(dt)` in the broadcast "loop":
```suggestion
        abs_dt = abs(dt)
        @.. WikRange = 1//2 * (W.dW * WikJ - (dwrange == i) * abs_dt) #+ (dwrange < i) * dt * WikJ2 - (dwrange > i) * dt * vec_χ)
```